apps/src/code-studio/components/progress/ProgressDot.jsx

```diff
@@ -253,18 +263,24 @@ export const ProgressDot = Radium(React.createClass({
          ]}
       >
         {(iconClassFromIconType[level.icon] && !isPeerReview) ?
-          <i
-            className={this.iconClassName()}
-            style={[
-              styles.dot.common,
-              styles.dot.puzzle,
-              courseOverviewPage && styles.dot.overview,
-              styles.dot.icon,
-              smallDot && styles.dot.icon_small,
-              status && status !== LevelStatus.not_tried && styles.dot.icon_complete,
-              outlineCurrent && {textShadow: createOutline(color.level_current)}
-            ]}
-          /> :
+          <span data-tip data-for={level.activeId} >
+            <i
+              className={this.iconClassName()}
+              style={[
+                styles.dot.common,
+                styles.dot.puzzle,
+                courseOverviewPage && styles.dot.overview,
+                styles.dot.icon,
+                smallDot && styles.dot.icon_small,
+                status && status !== LevelStatus.not_tried && styles.dot.icon_complete,
+                outlineCurrent && {textShadow: createOutline(color.level_current)}
+              ]}
+            />
+            <ReactTooltip id={level.activeId} aria-haspopup="true" role="example" delayShow={1000} wrapper="span">
```

Comment:
I will admit I know little about `aria-haspopup`, and the fact that you used it may indicate you know more than me. However, reading through this doc https://www.w3.org/TR/wai-aria/states_and_properties#aria-haspopup, it says
`Note that ordinary tooltips are not considered popups in this context.`